packages/ethereum-contracts/contracts/agreements/CFAv1Forwarder.sol

```diff
@@ -0,0 +1,493 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity >= 0.8.0;
+
+import {
+    ISuperfluid,
+    ISuperToken,
+    BatchOperation,
+    FlowOperatorDefinitions
+} from "../interfaces/superfluid/ISuperfluid.sol";
+
+import {
+    IConstantFlowAgreementV1
+} from "../interfaces/agreements/IConstantFlowAgreementV1.sol";
+
+import { CallUtils } from "../libs/CallUtils.sol";
+
+/**
+ * The CFAv1Forwarder contract provides an easy to use interface to
+ * ConstantFlowAgreementV1 specific functionality of Super Tokens.
+ * Instances of this contract can operate on the protocol only if configured as "trusted forwarder"
+ * by protocol governance.
+ */
+contract CFAv1Forwarder {
+
+    ISuperfluid internal _host;
+    IConstantFlowAgreementV1 internal _cfa;
+
+    // is tied to a specific instance of host and agreement contracts at deploy time
+    constructor(ISuperfluid host) {
+        _host = host;
+        _cfa = IConstantFlowAgreementV1(address(_host.getAgreementClass(
+            keccak256("org.superfluid-finance.agreements.ConstantFlowAgreement.v1")
+        )));
+    }
+
+    /**
+     * @notice Sets the given flowrate between msg.sender and a given receiver.
+     * If there's no pre-existing flow and `flowrate` non-zero, a new flow is created.
+     * If there's an existing flow and `flowrate` non-zero, the flowrate of that flow is updated.
+     * If there's an existing flow and `flowrate` zero, the flow is deleted.
+     * If the existing and given flowrate are equal, no action is taken.
+     * On creation of a flow, a "buffer" amount is automatically detracted from the sender account's available balance.
+     * This buffer is redeemed on deletion of that flow, if the sender account is then solvent (non-zero balance).
```

Comment:
a nitpick, but I think maybe you meant to say critical here? also, the account is not critical/solvent when their balance is at 0

Reply:
hmm, I think the statement is correct: buffer redeemed (to the flow sender) if sender account solvent (non-zero balance here implying it's positive). But will see if I can rephrase it to be clearer.

Reply:
ah okay gotcha